src/sync/sys/windows/net.rs

```diff
@@ -0,0 +1,303 @@
+            info!("listener shutdown for quit flag");
+            return Err(io::Error::new(
+                io::ErrorKind::Other,
+                "listener shutdown for quit flag",
+            ));
+        }
+
+        // Create a new pipe for every new client
+        let mut namedpipe = self.new_instance().unwrap();
+
+        let mut poll = Poll::new().unwrap();
+        let mut events = Events::with_capacity(1024);
+        poll.registry()
+            .register(
+                &mut namedpipe,
+                SERVER,
+                Interest::WRITABLE,
+            )
+            .unwrap();
+
+        loop {
+            match namedpipe.connect() {
+                Ok(()) => {
+                    // pipe is locked so can't use it here.
+                    break;
+                }
+                Err(e) if e.kind() == std::io::ErrorKind::WouldBlock => {
+                    println!("waiting for client to connect");
+                    poll.poll(&mut events, None).unwrap();
+                    continue;
+                }
+                Err(e) => {
+                    println!("Error connecting to pipe: {e}");
+                    return Err(e);
+                }
+            }
+        }
+
+        poll.registry()
+            .reregister(
+                &mut namedpipe,
+                SERVER,
+                Interest::READABLE,
+            )
+            .unwrap();
+        let instance_num = self.instance_number.fetch_add(1, Ordering::SeqCst);
+        trace!("pipe instance {} connected", instance_num);
+        let pipe_instance = PipeConnection {
+            named_pipe: Mutex::new(namedpipe),
+            poller: Mutex::new(poll),
+            instance_number: instance_num,
+        };
+        Ok(Some(pipe_instance))
+    }
+
+    fn new_instance(&self) -> io::Result<NamedPipe> {
+        let name = OsStr::new(&self.address.as_str())
+            .encode_wide()
+            .chain(Some(0)) // add NULL termination
+            .collect::<Vec<_>>();
+
+        // bitwise or file_flag_first_pipe_instance with file_flag_overlapped and pipe_access_duplex
+        let mut openmode = PIPE_ACCESS_DUPLEX | FILE_FLAG_OVERLAPPED;
+
+        if self.first_instance.load(Ordering::SeqCst) {
+            openmode |= FILE_FLAG_FIRST_PIPE_INSTANCE;
+            self.first_instance.swap(false, Ordering::SeqCst);
+        }
+
+        // Safety: syscall
+        let h = unsafe {
+            CreateNamedPipeW(
+                name.as_ptr(),
+                openmode,
+                PIPE_TYPE_BYTE,
+                PIPE_UNLIMITED_INSTANCES,
+                65536,
+                65536,
+                0,
+                std::ptr::null_mut(), // todo set this on first instance
+            )
+        };
+
+        if h == INVALID_HANDLE_VALUE {
+            Err(io::Error::last_os_error())
+        } else {
+            let pipe = unsafe { NamedPipe::from_raw_handle(h as RawHandle) };
+
+            Ok(pipe)
+        }
```

Comment:
Still.. rusty at rust, but would match be more idiomatic?

Reply:
makes sense to be idiomatic here.